tools/debian/first_package_finder.py

```diff
@@ -0,0 +1,120 @@
+"""Functions to help find the first package version for a specific release"""
+
+from urllib import request
+from datetime import datetime
+import gzip
+
+import pandas as pd
+
+
+def create_url(date: datetime, version: str = None):
+  """Create an url for snapshot.debian.org"""
+  base_url = 'https://snapshot.debian.org/archive/debian/{date}/dists/'
+  # `.gz` format always exist for all snapshots
+  source_path = '{version}/main/source/Sources.gz'
+
+  formatted_date = convert_datetime_to_str_datetime(date)
+
+  if version is None:
+    # return url for retrieving all versions for date
+    return base_url.format(date=formatted_date)
+  else:
+    return base_url.format(date=formatted_date) \
```

Comment:
nit: no need for an else after return.

Reply:
I do like having the if and the else as the last section of a function, I feel like it makes it clearer that these are the two return paths.

Reply:
Renamed to `get_debian_dists_url` and `get_debian_sources_url`